lucille-plugins/lucille-api/src/main/java/com/kmwllc/lucille/endpoints/LucilleResource.java

```diff
@@ -1,61 +1,211 @@
+    for (String configId : runnerManager.getConfigKeys  ()) {
+      ret.put(configId, runnerManager.getConfig(configId).root().unwrapped());
+    }
+    return Response.ok(ret).build();
+  }
 
-      RunStatus runStatus = new RunStatus(isRunning);
 
-      return Response.ok(runStatus).build();
-    } else {
-      return Response.status(401).build();
+  @GET
+  @Tag(name = "Config", description = "Retrieve Lucille configuration details.")
+  @Path("/config/{configId}")
+  @Operation(summary = "Get Lucille config",
+      description = "Retrieves a specific Lucille config by ID.",
+      security = @SecurityRequirement(name = "basicAuth"))
+  public Response getConfig(@Parameter(hidden = true) @Auth Optional<PrincipalImpl> user,
+      @Parameter(description = "The UUID of the configuration to retrieve.", required = false,
+          example = "fca83cb6-c2c2-4cbf-93ef-41c08d5d4b58") @PathParam("configId") String configId) {
+    Response authResponse = authHandler.authenticate(user);
+    if (authResponse != null) {
+      return authResponse; // Return if authentication fails
+    }
+
+    try {
+      Config config = runnerManager.getConfig(configId);
+      if (config == null) {
+        return StandardErrorResponse.buildResponse(Response.Status.NOT_FOUND,
+            "Configuration with ID " + configId + " not found.");
+      }
+      return Response.ok(config.root().unwrapped()).build();
+    } catch (Exception e) {
+      return StandardErrorResponse.buildResponse(Response.Status.INTERNAL_SERVER_ERROR,
+          "Error retrieving configuration: " + e.getMessage());
     }
   }
 
   @POST
-  public Response startRun(@Auth Optional<PrincipalImpl> user) {
+  @Tag(name = "Run")
+  @Path("/run")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Start a new Lucille run",
+      description = "Triggers a new Lucille run with the specified configuration if none is currently running.")
+  public Response startRun(@Parameter(hidden = true) @Auth Optional<PrincipalImpl> user,
+      @RequestBody(description = "Request body containing the configuration ID.", required = true,
+          content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(
+              type = "object",
+              example = "{\"configId\": \"550e8400-e29b-41d4-a716-446655440000\"}"))) Map<String, String> requestBody) {
+    Response authResponse = authHandler.authenticate(user);
+    if (authResponse != null) {
+      return authResponse; // Return if authentication fails
+    }
+
+    try {
+      // Extract configId from the request body
+      String configId = requestBody.get("configId");
+      if (configId == null || configId.isBlank()) {
+        return StandardErrorResponse.buildResponse(Response.Status.BAD_REQUEST,
+            "configId is required in the request body.");
+      }
 
-    if (user.isPresent()) {
-      boolean status = runnerManager.run();
+      String runId = UUID.randomUUID().toString();
```

Comment:
RunnerManager can generate a runId, right?
If so, we don't need to generate the ID here in API code.
The goal should be to minimize the number of different places where runIds get created and ideally confine it to Runner/RunnerManager only.

Reply:
It could, but RunManager gave the same option for something external to supply a runId.
In this case, its necessary because the contract is to get the RunDetails which this api consistently does for all started, and running "runs".   RunDetails are key'ed by RunId.  I didn't feel the RunManager needed to know about RunDetails, so in this case the runId is supplied externally.  Potentially it could be an optional parameter in the api .. I could imagine some test cases where this might be useful for full end-to-end, but didn't propagate it that far up, this pr.

Reply:
You wrote "I didn't feel the RunManager needed to know about RunDetails" but looking at RunnerManager I see many references to RunDetails so those classes look to me like they're intended to work together. 
RunnerManager.runWithConfig should be updated to return RunDetails instead of a boolean. The boolean made sense when RunDetails didn't exist, but now the RunDetails does exist, it the more natural return type for a method that launches a run. After this change, we'll be able to get the generated runId from the returned RunDetails and we don't have to generate the runId here and pass it in.

Reply:
Ok, that makes sense to me.  But this will generate a lot of signature changes, and pre-appologies .. I pressed the auto-format on this, so you can see what google standard format definition does in eclipse

Reply:
updated